Setting AppBundleDir to a custom value results in error MSB4044: The "SpotlightIndexer" task was not given a value for the required parameter "Input".

Repro: https://gist.github.com/rolfbjarne/68af146bb629e732e32e47125e7eb818

Result:

> error MSB4044: The "SpotlightIndexer" task was not given a value for the required parameter "Input".

Binlog:  [build-2022-05-12-175855.binlog.zip](https://github.com/xamarin/xamarin-macios/files/8680790/build-2022-05-12-175855.binlog.zip)

This is most likely the problem:

https://github.com/xamarin/xamarin-macios/blob/06041ab37f153c4e14f0e4a6036e8062a6d72d74/msbuild/Xamarin.Shared/Xamarin.Shared.targets#L2648-L2652

we don't set `_AppContainerDir` when `AppBundleDir` is set.

Workaround: set `_AppContainerDir` yourself. This should be the parent directory of `AppBundleDir`.

```xml
<PropertyGroup>
    <_AppContainerDir>$([System.IO.Path]::GetDirectoryName($(AppBundleDir)))/</_AppContainerDir>
</PropertyGroup>
```

The fix should be fairly simple: compute `_AppContainerDir` even if `AppBundleDir` is set.

Thanks @rolfbjarne, the error parse error goes away, but still have the:

` error MSB4044: The "SpotlightIndexer" task was not given a value for the required parameter "Input".`

does that property need to be in the .csproj? Or in the .targets?
I've it in the .csproj outside of the device condition but also tried within the one for Debug|iPhone for example and I keep getting this spotlight error.

--- a/msbuild/Xamarin.Shared/Xamarin.Shared.targets
+++ b/msbuild/Xamarin.Shared/Xamarin.Shared.targets
@@ -2682,6 +2682,9 @@ Copyright (C) 2018 Microsoft. All rights reserved.
 			<_AppContainerDir Condition="'$(IsAppDistribution)' == 'true'">$(ArchivePath)\Products\Applications\</_AppContainerDir>
 			<AppBundleDir>$(_AppContainerDir)$(_AppBundleName)$(AppBundleExtension)</AppBundleDir>
 		</PropertyGroup>
+		<PropertyGroup Condition="'$(_AppContainerDir)' == ''">
+			 <_AppContainerDir>$([System.IO.Path]::GetDirectoryName($(AppBundleDir)))/</_AppContainerDir>
+		</PropertyGroup>
 		<PropertyGroup>
 			<_AppBundlePath>$(AppBundleDir)\</_AppBundlePath>
 			<_AppResourcesRelativePath Condition="'$(_PlatformName)' == 'macOS' Or '$(_PlatformName)' == 'MacCatalyst'">Contents\Resources\</_AppResourcesRelativePath>

--- a/tests/dotnet/UnitTests/ProjectTest.cs
+++ b/tests/dotnet/UnitTests/ProjectTest.cs
@@ -982,5 +982,19 @@ namespace Xamarin.Tests {
 				Assert.That (errors [0].Message, Does.Contain ("Error loading Entitlements.plist template 'Entitlements.plist'"), "Message");
 			}
 		}
+
+		[TestCase (ApplePlatform.MacOSX, "osx-arm64")]
+		public void CustomAppBundleDir (ApplePlatform platform, string runtimeIdentifiers)
+		{
+			var project = "MySimpleApp";
+			Configuration.IgnoreIfIgnoredPlatform (platform);
+
+			var project_path = GetProjectPath (project, runtimeIdentifiers: runtimeIdentifiers, platform: platform, out var appPath);
+			Clean (project_path);
+			var properties = GetDefaultProperties (runtimeIdentifiers);
+			var customAppBundleDir = Path.Combine (Cache.CreateTemporaryDirectory (), project + ".app");
+			properties ["AppBundleDir"] = customAppBundleDir;
+			var result = DotNet.AssertBuild (project_path, properties);
+		}
 	}
 }

